plugins/package-managers/node/src/test/kotlin/pnpm/ModuleInfoTest.kt

```diff
@@ -0,0 +1,101 @@
+            val input = File("src/test/assets/pnpm-list.json").readText()
+
+            val expectedResults = listOf(
+                ModuleInfo(
+                    name = "some-project",
+                    version = "1.0.0",
+                    path = "/tmp/work/root",
+                    private = false,
+                    dependencies = mapOf(
+                        "eslint-scope" to ModuleInfo.Dependency(
+                            from = "eslint-scope",
+                            version = "link:node_modules/.pnpm/eslint-scope@5.1.1/node_modules/eslint-scope",
+                            path = "/tmp/work/root/node_modules/.pnpm/eslint-scope@5.1.1/node_modules/eslint-scope"
+                        )
+                    )
+                ),
+                ModuleInfo(
+                    name = "other-project",
+                    version = "1.0.1",
+                    path = "/tmp/work/other_root",
+                    private = false,
+                    dependencies = mapOf(
+                        "@types/eslint" to ModuleInfo.Dependency(
+                            from = "@types/eslint",
+                            version = "link:node_modules/.pnpm/@types+eslint@8.56.2/node_modules/@types/eslint",
+                            path = "/tmp/work/other_root/node_modules/.pnpm/@types+eslint@8.56.2" +
+                                "/node_modules/@types/eslint"
+                        )
+                    )
+                )
+            )
+
+            val moduleInfos = parsePnpmList(input)
+
+            moduleInfos.single() shouldContainExactly expectedResults
+        }
+
+        "handle multiple JSON arrays" {
+            val input = File("src/test/assets/pnpm-multi-list.json").readText()
+
+            val expectedResults = listOf(
+                listOf(
+                    ModuleInfo(
+                        name = "outer-project",
+                        version = "1.0.0",
+                        path = "/tmp/work/top",
+                        private = false,
+                        dependencies = mapOf(
+                            "eslint-scope" to ModuleInfo.Dependency(
+                                from = "eslint-scope",
+                                version = "link:node_modules/.pnpm/eslint-scope@5.1.1/node_modules/eslint-scope",
+                                path = "/tmp/work/top/node_modules/.pnpm/eslint-scope@5.1.1/node_modules/eslint-scope"
+                            )
+                        )
+                    )
+                ),
+                listOf(
+                    ModuleInfo(
+                        name = "nested-project",
+                        version = "1.0.0",
+                        path = "/tmp/work/top/nested",
+                        private = false
+                    )
+                )
+            )
+
+            val moduleInfos = parsePnpmList(input)
+
+            moduleInfos.toList() shouldContainExactly expectedResults
```

Comment:
...you can omit the `.toList()` conversion here and `import io.kotest.matchers.sequences.shouldContainExactly` instead of the collection version.

Reply:
Changed. Both test cases now operate on sequences.

Reply:
Damn, looks like this now leads to ["This sequence can be consumed only once"](https://github.com/oss-review-toolkit/ort/actions/runs/13070542617/job/36471025785?pr=9863#step:7:2814)?

Reply:
Yes, switched back to returning a `List`.

Reply:
I probably lost track a bit, but I'm surprised that the test did not need to change again. Anyway, I'll approve 😄

Reply:
Ah, that's because `parsePnpmList()` is tested, not `listModules()`.